metrix-integration/src/main/java/com/powsybl/metrix/integration/chunk/ChunkCutter.java

```diff
@@ -9,43 +9,46 @@
 
 import com.google.common.collect.Range;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Paul Bui-Quang {@literal <paul.buiquang at rte-france.com>}
  */
 public class ChunkCutter {
 
-    private final int firstVariant;
-
-    private final int lastVariant;
-
     private final int chunkSize;
 
-    public ChunkCutter(int firstVariant, int lastVariant, int chunkSize) {
-        if (firstVariant < 0) {
-            throw new IllegalArgumentException("First variant (" + firstVariant + ") has to be positive");
-        }
-        if (lastVariant < firstVariant) {
-            throw new IllegalArgumentException("Last variant (" + lastVariant +
-                    ") has to be greater or equals to first variant (" + firstVariant + ")");
-        }
+    private final List<Range<Integer>> ranges = new ArrayList<>();
+
+    public ChunkCutter(List<Range<Integer>> ranges, int chunkSize) {
```

Comment:
You can probably leave the old constructor:

```suggestion
    public ChunkCutter(int firstVariant, int lastVariant, int chunkSize) {
        this(List.of(Range.closed(firstVariant, lastVariant)), chunkSize);
    }

    public ChunkCutter(List<Range<Integer>> ranges, int chunkSize) {
```

Reply:
If you choose to do so, you can revert a change in one of the ChunkCutterTest to keep the use of the old constructor.

Reply:
done